src/compiler/types.ts

```diff
@@ -7429,6 +7445,7 @@ export const enum ScriptTarget {
     ES2021 = 8,
     ES2022 = 9,
     ES2023 = 10,
+    ES2024 = 11,

```

Comment:
This actually shouldn't be added yet; 2024 isn't finalized. Anything that refers to ES2024 should either be removed or point to ESNext

Reply:
Pushed a change for this (we're trying to get this in for 5.5!)

Reply:
Afterwards, we should move this enum to `LanguageFeatureMinimumTarget`, which is the canonical place to store language feature values so we know what to update when ratification happens.